manager/controllers/app/fybrikapplication_controller.go

```diff
@@ -553,24 +562,27 @@ func (r *FybrikApplicationReconciler) constructDataInfo(req *datapath.DataInfo,
 	configEvaluatorInput.Request = CreateDataRequest(input, req.Context, &req.DataDetails.ResourceMetadata)
 
 	// Governance actions
-	err = r.checkGovernanceActions(configEvaluatorInput, req, appContext, env)
+	governanceMsg, err = r.checkGovernanceActions(configEvaluatorInput, req, appContext, env)
 	if err != nil {
-		return err
+		return "", errors.WithMessage(err, governanceMsg)
 	}
 	configDecisions, err := r.ConfigEvaluator.Evaluate(configEvaluatorInput)
 	if err != nil {
 		appContext.Log.Error().Err(err).Msg("Error evaluating config policies")
-		return err
+		return "", err
 	}
 	logging.LogStructure("Config Policy Decisions", configDecisions, appContext.Log, zerolog.DebugLevel, false, false)
 	req.WorkloadCluster = configEvaluatorInput.Workload.Cluster
 	req.Configuration = configDecisions
-	return nil
+	// propagate messages from the catalog and policy manager
+	msg := strings.TrimPrefix(catalogMsg+Separator+governanceMsg, Separator)
```

Comment:
Note:
What is the message's format? If it is a "closed" JSON, I mean 
```json
{
	"code": "FYBMG0001W",
	"level": "WARN / INFO / ERROR",
	"description": "some message description"
}
```
 we will get  something like:
 ```json
{
	"code": "FYBMG0001W",
	"level": "WARN / INFO / ERROR",
	"description": "some message description"
};
{
	"code": "FYBMG0001W",
	"level": "WARN / INFO / ERROR",
	"description": "some message description"
}
```
Which is not a valid JSON, on the other hand the message is defined as a String, so on the Fybrik side it will work.

Reply:
For Fybrik, it is a generic string.